File: test-platform-v2/frontend/src/api/aiOperations.ts
import { aitdeV2 } from './missions'

/** AIOperationRecord as exposed by GET /api/v2/ai-operations (V30-084/085). */
export interface AiOperation {
  id: number
  project_id: number
  mission_id: number
  operation_type: string
  status: string
  model_provider: string
  model_name: string
  prompt_version: string
  schema_version: string
  result_ref_json: string
  error_code: string
  error_message: string
  duration_ms: number
  token_usage_json: string
  created_at: string | null
  finished_at: string | null
}

/**
 * List a mission's AI operation records (AI Debug Drawer feed, V30-085).
 * Requires the `mission:ai_view_debug` permission.
 */
export function fetchAiOperations(missionId: number, signal?: AbortSignal): Promise<AiOperation[]> {
  return aitdeV2
    .get('/ai-operations', { params: { mission_id: missionId }, signal })
    .then((res) => (res as { items?: AiOperation[] } | undefined)?.items ?? [])
}

export const AI_OPERATION_STATUS_LABELS: Record<string, string> = {
  QUEUED: '排队中',
  RUNNING: '执行中',
  SUCCEEDED: '成功',
  FAILED: '失败',
  CANCELLED: '已取消',
}

File: test-platform-v2/backend/app/api/v2/ai_operations.py
"""AITDE v2 AI Operation API (V30-084)."""
from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.v2.deps import require_aitde_v3
from app.core.deps import CurrentUser, get_db, require_permission
from app.modules.aitde.ai_ops import service
from app.modules.aitde.ai_ops.models import AIOperationRecord
from app.schemas.common import R

router = APIRouter(
    prefix="/ai-operations",
    tags=["AITDE - AI Governance"],
    dependencies=[Depends(require_aitde_v3)],
)


@router.get("/{operation_id}", response_model=R[dict])
def get_operation(
    operation_id: int,
    current: CurrentUser = Depends(require_permission("mission:detail")),
    db: Session = Depends(get_db),
):
    record = service.get_operation(db, operation_id)
    return R.ok(_to_dict(record))


def _to_dict(row: AIOperationRecord) -> dict:
    return {
        "id": row.id,
        "project_id": row.project_id,
        "mission_id": row.mission_id,
        "operation_type": row.operation_type,
        "status": row.status,
        "model_provider": row.model_provider,
        "model_name": row.model_name,
        "prompt_version": row.prompt_version,
        "schema_version": row.schema_version,
        "result_ref_json": row.result_ref_json,
        "error_code": row.error_code,
        "error_message": row.error_message,
        "duration_ms": row.duration_ms,
        "created_at": row.created_at.isoformat() if row.created_at else None,
        "finished_at": row.finished_at.isoformat() if row.finished_at else None,
    }
